plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -408,6 +450,96 @@ func (s *SQLServer) Init() error {
 	return nil
 }
 
+// Get Token Provider by loading cached token or refreshed token
+func getTokenProvider() (func() (string, error), error) {
```

Comment:
Do you think it is better to move token related methods to a new class? That would help to keep this main class minimal.